### PR TITLE
update to Singular_jll v403.214.1400

### PR DESCRIPTION
**Pull request by hannes14**

### CODE REVIEW

File: deps/src/ideals.cpp [hunk 1]
```diff
@@ -100,12 +100,16 @@ auto id_prune_map_helper(sip_sideal * I, ring R)
   return std::make_tuple(s, TT);
 }
 
-auto id_prune_map_v_helper(sip_sideal * I, int* v, ring R)
+auto id_prune_map_v_helper(sip_sideal * I, jlcxx::ArrayRef<int> a, ring R)
 {
   auto origin = currRing;
   rChangeCurrRing(R);
   ideal      T;
+  int *v=omAlloc(I->rank*siseof(int));
```
Comment (fingolfin) [inline deps/src/ideals.cpp:108]:
```suggestion
  int *v=omAlloc(I->rank*sizeof(int));
```

**Comment by fingolfin** [comment]
@hannes14 just for the record, you can test these changes (in particular to see if they have syntax errors) locally... just `dev` the package and do `using Singular`

**Comment by fingolfin** [comment]
Oscar.jl tests fail, but I hope it's a fluke... CC @ThomasBreuer  we see this in [the logs](https://github.com/oscar-system/Singular.jl/actions/runs/7575395438/job/20631939161?pr=748):
```
┌ Error: doctest failure in ~/.julia/packages/Oscar/1Moid/src/Groups/libraries/atlasgroups.jl:276-285
│ 
│ ```jldoctest
│ julia> number_atlas_groups("A5")
│ 18
│ 
│ julia> number_atlas_groups(PermGroup, "A5")
│ 3
│ 
│ julia> number_atlas_groups(MatrixGroup, "A5")
│ 15
│ 
│ ```
│ 
│ Subexpression:
│ 
│ number_atlas_groups("A5")
│ 
│ Evaluated output:
│ 
│ ERROR: Error thrown by GAP: Error, FuncAPPEND_LIST_INTR: <list2> must be a small list (not a character) in
│   Append( res, arg[i] ); at /home/runner/.julia/artifacts/b5c2f0f824457e5c391fb24916f94d5d91c62c4f/share/gap/lib/list.gi:2086 called from
│ Concatenation( JoinStringsWithSeparator( tocid, "|" ), "|", gapname ) at /home/runner/.julia/scratchspaces/c863536a-3901-11e9-33e7-d5cd0df7b904/gap_4855902246029889253_1.9/pkg/atlasrep/gap/interfac.gi:897 called from
│ AGR.MergedTableOfContents( tocid, gapname ) at /home/runner/.julia/scratchspaces/c863536a-3901-11e9-33e7-d5cd0df7b904/gap_4855902246029889253_1.9/pkg/atlasrep/gap/interfac.gi:1241 called from
│ AGR.AtlasGeneratingSetInfo( arg, "all" ) at /home/runner/.julia/scratchspaces/c863536a-3901-11e9-33e7-d5cd0df7b904/gap_4855902246029889253_1.9/pkg/atlasrep/gap/interfac.gi:1324 called from
│ <function "AllAtlasGeneratingSetInfos">( <arguments> )
│  called from read-eval loop at *defin*:0
│ 
│ Stacktrace:
│  [1] error(s::String)
│    @ Base ./error.jl:35
│  [2] ThrowObserver(depth::Int32)
│    @ GAP ~/.julia/packages/GAP/aJO9M/src/GAP.jl:89
│  [3] _call_gap_func
│    @ ~/.julia/packages/GAP/aJO9M/src/ccalls.jl:325 [inlined]
│  [4] (::GAP.GapObj)(a1::GAP.GapObj)
│    @ GAP ~/.julia/packages/GAP/aJO9M/src/ccalls.jl:302
│  [5] number_atlas_groups(name::String)
│    @ Oscar ~/.julia/packages/Oscar/1Moid/src/Groups/libraries/atlasgroups.jl:289
│  [6] top-level scope
│    @ none:1
│ 
```


I'll restart that CI job

**Comment by ederc** [comment]
@fingolfin I looked into this error, but I could not see any connection to this PR. Let's see what happens for the restarted checks.

**Comment by ThomasBreuer** [comment]
Concerning the error messages, I have no idea yet what happens:
`JoinStringsWithSeparator` has been successful, and `Concatenation` would output the shown error message if the argument `gapname` would be a character. But how can I get into this situation by calling `AGR.AtlasGeneratingSetInfo`?

**Review by ederc — APPROVED** [review]

**Pull request merged**